pkg/gardenlet/operation/botanist/advertisedaddresses.go

```diff
@@ -82,5 +88,59 @@ func (b *Botanist) ToAdvertisedAddresses() ([]gardencorev1beta1.ShootAdvertisedA
 		})
 	}
 
+	ingressItems, err := b.GetIngressAdvertisedEndpoints(ctx)
+	if err != nil {
+		return nil, err
+	}
+	addresses = append(addresses, ingressItems...)
+
 	return addresses, nil
 }
+
+// GetIngressAdvertisedEndpoints returns a list of
+// [gardencorev1beta1.ShootAdvertisedAddress] items, which have been derived
+// from any existing [networkingv1.Ingress] resources labeled with
+// [v1beta1constants.LabelShootEndpointAdvertise].
+func (b *Botanist) GetIngressAdvertisedEndpoints(ctx context.Context) ([]gardencorev1beta1.ShootAdvertisedAddress, error) {
+	result := make([]gardencorev1beta1.ShootAdvertisedAddress, 0)
+	var ingressList networkingv1.IngressList
+
+	err := b.SeedClientSet.Client().List(
+		ctx,
+		&ingressList,
+		client.InNamespace(b.Shoot.GetInfo().Status.TechnicalID),
+		client.MatchingLabels(map[string]string{
+			v1beta1constants.LabelShootEndpointAdvertise: "true",
+		}),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// Only the TLS items are processed, since
+	// [gardencorev1beta1.ShootAdvertisedAddress] is constrained to https://
+	// endpoints only.
```

Comment:
```suggestion
	ingressList := &networkingv1.IngressList{}
	if err := b.SeedClientSet.Client().List(
		ctx,
		ingressList,
		client.InNamespace(b.Shoot.GetInfo().Status.TechnicalID),
		client.MatchingLabels(map[string]string{
			v1beta1constants.LabelShootEndpointAdvertise: "true",
		}),
	); err != nil {
		return nil, err
	}

	// Only the TLS items are processed, since
	// [gardencorev1beta1.ShootAdvertisedAddress] is constrained to https://
	// endpoints only.
	result := make([]gardencorev1beta1.ShootAdvertisedAddress, 0)
```

Nit: Minor style/convention

Reply:
Fixed.